### DISCUSSION TITLE
detekt works as expected only after clean project was made

### DISCUSSION THREAD

**Discussion by Den-Rimus** [detekt/detekt#3397] [Q&A] [answered]
I'm running detekt using `./gradlew detektMain` command in terminal.
It only runs fine and detects issues after a "clean project" was made. In this case it runs for 2.5 minutes and finds lots of issues(`ignoreFailures` is set to `true`).
If I run it two times consequently - second run just gives me BUILD SUCCESSFUL after 7 seconds and no issues detected.

Configuration:
```
plugins {
    id("io.gitlab.arturbosch.detekt").version("1.15.0")
```

also:
```
subprojects {
    apply {
        plugin("io.gitlab.arturbosch.detekt")
    }

    subproject.tasks.withType<io.gitlab.arturbosch.detekt.Detekt>().configureEach {
        // Target version of the generated JVM bytecode. It is used for type resolution.
        this.jvmTarget = "1.8"
    }

    dependencies {
        "detektPlugins"("io.gitlab.arturbosch.detekt:detekt-formatting:1.15.0")
    }

    detekt {
        config = files("$rootDir/detekt/config.yml")
        autoCorrect = isLocalDevMachine // add `IS_LOCAL_DEV_MACHINE=true` to ~/.gradle/gradle.properties file
        ignoreFailures = true
        ignoredBuildTypes = listOf("release")
        ignoredFlavors = listOf("snapshot")
    }
```

Can it be configured to run every time?

**Comment by cortinico** [comment, marked as answer]
> Can it be configured to run every time?

Yes, you can achieve this using the `--rerun-tasks` flag when running detekt so `./gradlew detektMain --rerun-tasks`.
That would have the side effect of rerunning all the tasks, not only detekt (so can be quite expensive).

As an alternative, you should be able to update your `build.gradle.kts` you posted with:

```kotlin
tasks.withType<io.gitlab.arturbosch.detekt.Detekt>().configureEach {
        outputs.upToDateWhen { false }
}
```

That will cause only Detekt to re-run every time.

As a side note, please note that this is the intended behavior. Detekt doesn't re-run as the analysis was already executed and you have the reports already available in the build folder.

**Reply by Den-Rimus** [reply]
Somehow 'alternative' approach did not work. But taking that this is desired behaviour I will mark this as answer.

Big thanks for help - and on a such short notice!

**Comment by picklebento** [comment]
You seem to have `autoCorrect = true` when running locally. Did the first run autocorrects the findings?

**Reply by Den-Rimus** [reply]
Yes, of course.
